test/gtest/fusion_test.cpp

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+
+#include "tensor_holder.hpp"
+#include "get_handle.hpp"
+#include "cba.hpp"
+
+#if MIOPEN_BACKEND_HIP
+
+template <typename T>
+class FusionSetArgTest : public ConvBiasActivInferTest<T>
+{
+public:
+    void SetUp() override
+    {
+        cba<T>::SetUp();
+        weights2 =
+            tensor<T>{miopen_type<T>{}, cba<T>::tensor_layout, cba<T>::conv_config.GetWeights()};
+        weights2.generate(tensor_elem_gen_integer{3});
+        cba<T>::weights = weights2;
+        auto&& handle   = get_handle();
+        cba<T>::wei_dev = handle.Write(weights2.data);
+        handle.Finish();
+    }
+
+    void TearDown() override { cba<T>::TearDown(); }
+
+    template <typename Tp>
+    using cba = ConvBiasActivInferTest<Tp>;
+
+    tensor<T> weights2;
+    miopen::Allocator::ManageDataPtr wei_dev2;
+};
+
+struct FusionSetArgTestFloat : FusionSetArgTest<float>
+{
+};
+
+TEST_P(FusionSetArgTestFloat, TestSetArgApiCall)
+{
+    // Original fusion_plan/args execution happens in cba_infer.cpp
+    // Original is checked independently and not sequentially, prior to FusionTestSetArgTest.
+    using cba_float = cba<float>;
```

Comment:
There is no check of the datatype under the test. As a result, this test only works with FP32, no matter what data type is specified. Please see how we do observe `MIOPEN_TEST_FLOAT_ARG` and use it in google tests in #2193. Thank you.

Reply:
Hi @gonidelis could you fix in a follow up PR?